android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityAndroidPayFragment.java

```diff
@@ -0,0 +1,172 @@
+
+        UnityAndroidPayFragmentBuilder setCountryCode(String countryCode) {
+            bundle.putString(EXTRA_COUNTRY_CODE, countryCode);
+            return this;
+        }
+
+        UnityAndroidPayFragmentBuilder setEnvironment(int environment) {
+            bundle.putInt(EXTRA_ANDROID_PAY_ENVIRONMENT, environment);
+            return this;
+        }
+
+        UnityAndroidPayFragmentBuilder setPublicKey(String key) {
+            bundle.putString(EXTRA_PUBLIC_KEY, key);
+            return this;
+        }
+
+        UnityAndroidPayFragment build() {
+            checkBundleContainsRequiredExtras();
+            
+            UnityAndroidPayFragment fragment = new UnityAndroidPayFragment();
+            fragment.setArguments(bundle);
+            return fragment;
+        }
+
+        private void checkBundleContainsRequiredExtras() {
+            List<String> missingExtras = new ArrayList<>();
+            for (String extra : requiredExtras) {
+                if (bundle.get(extra) == null) {
+                    missingExtras.add(extra);
+                }
+            }
+
+            if (missingExtras.size() > 0) {
+                throw new IllegalArgumentException("Missing required bundle extras: " + missingExtras.toString());
+            }
+        }
+    }
+
+    public static UnityAndroidPayFragmentBuilder builder() {
+        return new UnityAndroidPayFragmentBuilder();
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        Bundle bundle = getArguments();
+        if (bundle != null) {
+            unityDelegateObjectName = bundle.getString(EXTRA_UNITY_DELEGATE_OBJECT_NAME);
+            cart = bundle.getParcelable(EXTRA_PAY_CART);
+            countryCode = bundle.getString(EXTRA_COUNTRY_CODE);
+            androidPayEnvironment = bundle.getInt(EXTRA_ANDROID_PAY_ENVIRONMENT,
+                    WalletConstants.ENVIRONMENT_TEST);
+            androidPublicKey = bundle.getString(EXTRA_PUBLIC_KEY);
+        }
+
+        // Don't recreate this fragment during configuration change.
+        setRetainInstance(true);
+
+        googleApiClient = new GoogleApiClient.Builder(this.getActivity())
+                .addApi(Wallet.API, new Wallet.WalletOptions.Builder()
+                        .setEnvironment(androidPayEnvironment)
+                        .setTheme(WalletConstants.THEME_DARK)
+                        .build())
+                .addConnectionCallbacks(this)
+                .build();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        googleApiClient.connect();
+    }
+
+    @Override
+    public void onStop() {
+        googleApiClient.disconnect();
+        super.onStop();
+    }
+
+    @Override
+    public void onConnected(@Nullable Bundle bundle) {
+        // On connection, request for a masked wallet.
+        PayHelper.requestMaskedWallet(googleApiClient, cart, androidPublicKey);
```

Comment:
Is this right behaviour, every time when we connected to play services we will request masked wallet. Keep in mind that this will happen I guess even in case when user presses home button.
Activity will go to background `onStop` will be called, `googleApiClient.disconnect` will be called as well. Then user brings app back, `onStart()` -> `googleApiClient.connect()` -> `PayHelper.requestMaskedWallet(googleApiClient, cart, androidPublicKey)`.

Just asking if this flow is expected.